### PR TITLE
[FEDRAMP-629, IDENTITY-2574] Okta SSO login

### PR DESCRIPTION
**Pull request by brianstrauch**
Release Notes
-------------
New Features
- PLACEHOLDER

Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * no: DO NOT MERGE until the required features are live in prod  

What
----
Only supports login for `confluent login --url https://infra.confluentgov-internal.com`

Test & Review
-------------
Mostly manual testing for now.

### CODE REVIEW

File: internal/pkg/auth/sso/auth_state.go [hunk 1]
```diff
@@ -63,6 +75,8 @@ type authState struct {
 	SSOProviderClientID           string
 	SSOProviderCallbackUrl        string
 	SSOProviderIdentifier         string
+	SSOProviderScope              string
+	SSOProviderIDP                string
```
Comment (sgagniere) [inline internal/pkg/auth/sso/auth_state.go:79]: I see the IDP being set, but I'm not sure where it's used.

File: internal/pkg/auth/auth_token_handler.go [hunk 2]
```diff
@@ -80,23 +81,34 @@ func (a *AuthTokenHandlerImpl) GetCCloudTokens(clientFactory CCloudClientFactory
...
 
 func (a *AuthTokenHandlerImpl) getCCloudSSOToken(client *ccloudv1.Client, noBrowser bool, email, orgResourceId string) (string, string, error) {
-	userSSO, err := a.getCCloudUserSSO(client, email, orgResourceId)
-	if err != nil {
-		log.CliLogger.Debugf("unable to obtain user SSO info: %v", err)
-		return "", "", errors.Errorf(errors.FailedToObtainedUserSSOErrorMsg, email)
-	}
-	if userSSO == "" {
-		return "", "", errors.Errorf(errors.NonSSOUserErrorMsg, email)
+	isOkta := types.Contains([]string{"fedramp", "fedramp-internal"}, sso.GetCCloudEnvFromBaseUrl(client.BaseURL))
```
[the remark this answers is not in the document]
Comment (brianstrauch) [reply internal/pkg/auth/auth_token_handler.go:84]: I can add "fedramp" to this list when it's ready!
```suggestion
	isOkta := sso.GetCCloudEnvFromBaseUrl(client.BaseURL) == "fedramp-internal"
```

File: internal/pkg/ccloudv2/utils.go [hunk 3]
```diff
@@ -24,7 +24,12 @@ type NullableString interface {
 	IsSet() bool
 }
 
-var Hostnames = []string{"confluent.cloud", "cpdev.cloud"}
+var Hostnames = []string{
+	"confluent.cloud",
+	"confluentgov-internal.com",
+	"confluentgov.com",
```
Comment (brianstrauch) [inline internal/pkg/ccloudv2/utils.go:30]:
```suggestion
```

File: internal/pkg/auth/sso/auth_state.go (hunk 1 above)
Comment (brianstrauch) [reply internal/pkg/auth/sso/auth_state.go:79]: It's defined here: https://github.com/confluentinc/cli/pull/1871/files#diff-796fc20ed244a970061f37913241dbb8c512cfff8a27fbc3391f9b1bccbe3035R36
and used here: https://github.com/confluentinc/cli/pull/1871/files#diff-796fc20ed244a970061f37913241dbb8c512cfff8a27fbc3391f9b1bccbe3035R115

**Review by sgagniere — APPROVED** [review]
+1, thanks

Is the `no: DO NOT MERGE until the required features are live in prod` outdated in the description?

Also, no release notes because it's currently just internal right?

**Comment by brianstrauch** [comment]
> Is the no: DO NOT MERGE until the required features are live in prod outdated in the description?

No, still waiting for the backend on a small issue around refresh tokens

> Also, no release notes because it's currently just internal right?
:+1:

File: internal/pkg/auth/auth_token_handler.go [hunk 4]
```diff
@@ -80,23 +81,34 @@ func (a *AuthTokenHandlerImpl) GetCCloudTokens(clientFactory CCloudClientFactory
...
+			return "", "", errors.Errorf(errors.FailedToObtainedUserSSOErrorMsg, email)
+		}
+		if userSSO == "" {
+			return "", "", errors.Errorf(errors.NonSSOUserErrorMsg, email)
+		}
+		auth0ConnectionName = userSSO
 	}
 
-	idToken, refreshToken, err := sso.Login(client.BaseURL, noBrowser, userSSO)
+	idToken, refreshToken, err := sso.Login(client.BaseURL, noBrowser, auth0ConnectionName)
```
Comment (MuweiHe) [inline internal/pkg/auth/auth_token_handler.go:99]: `auth0ConnectionName` would be empty for fedramp?

File: internal/pkg/auth/sso/auth0_utils.go [hunk 5]
```diff
@@ -8,19 +8,20 @@ import (
...
 
 var auth0ClientIds = map[string]string{
-	"prod":  "oX2nvSKl5jvBKVgwehZfvR4K8RhsZIEs",
-	"stag":  "8RxQmZEYtEDah4MTIIzl4hGGeFwdJS6w",
-	"devel": "sPhOuMMVRSFFR7HfB606KLxf1RAU4SXg",
-	"cpd":   "7rG4pmRbnMn5mIsEBLAP941IE1x2rNqC",
-	"test":  "00000000000000000000000000000000",
+	"cpd":              "7rG4pmRbnMn5mIsEBLAP941IE1x2rNqC",
+	"devel":            "sPhOuMMVRSFFR7HfB606KLxf1RAU4SXg",
+	"fedramp-internal": "0oa7c9gkc6bHBD2OW1d7",
```
Comment (MuweiHe) [inline internal/pkg/auth/sso/auth0_utils.go:13]: will there be an id for `fedramp`?

Comment (brianstrauch) [reply internal/pkg/auth/sso/auth0_utils.go:13]: Yes, but in a separate PR!

File: internal/pkg/auth/auth_token_handler.go (hunk 4 above)
Comment (brianstrauch) [reply internal/pkg/auth/auth_token_handler.go:99]: Yes, that's intended! It won't get added to the API call if it's `""`.

**Pull request merged**